flink-runtime/src/main/java/org/apache/flink/runtime/operators/Driver.java

```diff
@@ -87,4 +87,6 @@
 	 * @throws Exception Exceptions may be forwarded.
 	 */
 	void cancel() throws Exception;
+
+	void resetForIterativeTasks() throws Exception;
```

Comment:
Currently, the `ResettableDriver` interface extends the `Driver` interface. I would say that for the moment, let's just start by making some of the drivers resettable in this PR, and keep both interfaces around. And then when this PR is merged, a second PR could make all the remaing drivers resettable as well. And then a third one could merge the functionality of `ResettableDriver` into `Driver`, and thus get rid of `ResettableDriver`.